pkg/controller/deployment/deployment_controller.go

```diff
@@ -557,9 +597,23 @@ func (dc *DeploymentController) getNewRC(deployment extensions.Deployment) (*api
 		dc.rcExpectations.DeleteExpectations(dKey)
 		return nil, fmt.Errorf("error creating replication controller: %v", err)
 	}
+
+	if err = dc.updateDeploymentVersion(deployment, newVersion); err != nil {
+		return createdRC, err
+	}
+
 	return createdRC, nil
 }
 
+func (dc *DeploymentController) updateDeploymentVersion(deployment extensions.Deployment, version string) error {
+	if deployment.Annotations == nil {
+		deployment.Annotations = make(map[string]string)
+	}
+	deployment.Annotations[deploymentutil.VersionAnnotation] = version
```

Comment:
Disadvantages of using generation for this purpose:
* generation needs to be incremented when changes are made that won't cause rollouts, such as scaling (changing replicas)
* generation would reset to 0 after deleting and re-creating the Deployment